include/ginkgo/core/base/utils.hpp

```diff
@@ -517,12 +530,14 @@ class temporary_clone {
     {
         if (ptr->get_executor() == exec) {
             // just use the object we already have
-            handle_ = handle_type(ptr, [](pointer) {});
+            handle_ = handle_type(ptr, null_deleter<T>());
         } else {
             // clone the object to the new executor and make sure it's copied
             // back before we delete it
-            handle_ = handle_type(gko::clone(std::move(exec), ptr).release(),
-                                  copy_back_deleter<T>(ptr));
+            handle_ = handle_type(
+                temporary_clone_helper<T>::create(std::move(exec), ptr)
+                    .release(),
```

Comment:
Why using the helper here? It looks to me like clone already gave the proper type?

Reply:
clone only works for polymorphic types, I need it to work for Array as well, where we involve a constructor instead of calling `object->copy_from`